src/julia.h

```diff
@@ -87,7 +92,7 @@ DLLEXPORT void jl_threading_profile(void);
 #  define JL_ATOMIC_FETCH_AND_ADD(a,b)                                    \
        _InterlockedExchangeAdd((volatile LONG *)&(a), (b))
 #  define JL_ATOMIC_COMPARE_AND_SWAP(a,b,c)                               \
-       _InterlockedCompareExchange64(&(a), (c), (b))
+       _InterlockedCompareExchange64((volatile LONG64 *)&(a), (c), (b))
```

Comment:
looks like I can get rid of the error and case here for at least COMPARE_AND_SWAP by changing `JL_DEFINE_MUTEX` (and `JL_DEFINE_MUTEX_EXT`) to use `int64_t volatile m` instead of `uint64_t volatile m`. Is that okay on unix?